amazon-kinesis-client-multilang/src/test/java/software/amazon/kinesis/multilang/MultiLangDaemonConfigTest.java

```diff
@@ -35,62 +31,168 @@
-        ClassLoader classLoader = Mockito.mock(ClassLoader.class);
+    public void testConstructorUsingStreamName() throws IOException {
+        setup(TEST_STREAM_NAME, null, TEST_REGION);
+
+        assertConfigurationsMatch(TEST_STREAM_NAME, null);
+    }
+
+    @Test
+    public void testConstructorUsingStreamNameAndStreamArnIsEmpty() throws IOException {
+        setup(TEST_STREAM_NAME, "", TEST_REGION);
+
+        assertConfigurationsMatch(TEST_STREAM_NAME, "");
+    }
+
+    @Test
+    public void testConstructorUsingStreamNameAndStreamArnIsWhitespace() throws IOException {
+        setup(TEST_STREAM_NAME, "   ", TEST_REGION);
+
+        assertConfigurationsMatch(TEST_STREAM_NAME, "");
+    }
 
-        Mockito.doReturn(new ByteArrayInputStream(PROPERTIES.getBytes())).when(classLoader)
-                .getResourceAsStream(FILENAME);
+    @Test
+    public void testConstructorUsingStreamArn() throws IOException {
+        setup(null, TEST_STREAM_ARN, TEST_REGION);
+
+        assertConfigurationsMatch(TEST_STREAM_NAME_IN_ARN, TEST_STREAM_ARN);
+    }
 
-        MultiLangDaemonConfig deamonConfig = new MultiLangDaemonConfig(FILENAME, classLoader, configurator);
+    @Test
+    public void testConstructorUsingStreamNameAsEmptyAndStreamArn() throws IOException {
+        setup("", TEST_STREAM_ARN, TEST_REGION);
+
+        assertConfigurationsMatch(TEST_STREAM_NAME_IN_ARN, TEST_STREAM_ARN);
+    }
 
+    @Test
+    public void testConstructorUsingStreamArnOverStreamName() throws IOException {
+        setup(TEST_STREAM_NAME, TEST_STREAM_ARN, TEST_REGION);
+
+        assertConfigurationsMatch(TEST_STREAM_NAME_IN_ARN, TEST_STREAM_ARN);
+    }
+
+    /**
+     * Verify the daemonConfig properties are what we expect them to be.
+     * @param deamonConfig
+     * @param expectedStreamName
+     */
+    private void assertConfigurationsMatch(String expectedStreamName,
+                                           String expectedStreamArn){
         assertNotNull(deamonConfig.getExecutorService());
         assertNotNull(deamonConfig.getMultiLangDaemonConfiguration());
         assertNotNull(deamonConfig.getRecordProcessorFactory());
+
+        assertEquals(TEST_EXE, deamonConfig.getRecordProcessorFactory().getCommandArray()[0]);
+        assertEquals(TEST_APPLICATION_NAME, deamonConfig.getMultiLangDaemonConfiguration().getApplicationName());
+        assertEquals(expectedStreamName, deamonConfig.getMultiLangDaemonConfiguration().getStreamName());
+        assertEquals(TEST_REGION, deamonConfig.getMultiLangDaemonConfiguration().getDynamoDbClient().get("region").toString());
+        assertEquals(TEST_REGION, deamonConfig.getMultiLangDaemonConfiguration().getCloudWatchClient().get("region").toString());
+        assertEquals(TEST_REGION, deamonConfig.getMultiLangDaemonConfiguration().getKinesisClient().get("region").toString());
+        assertEquals(expectedStreamArn, deamonConfig.getMultiLangDaemonConfiguration().getStreamArn());
```

Comment:
1. For legibility, suggest capturing `dC.getMLDC` as a local variable.
2. Per the primitive obsession call-out earlier, using a strongly-typed object would eliminate the `region.toString()` which helps made the code simpler (*and* stronger-typed!).